mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipperTest.java

```diff
@@ -0,0 +1,117 @@
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup;
+
+import static org.apache.james.mailbox.backup.MailboxMessageFixture.MESSAGE_1;
+import static org.apache.james.mailbox.backup.MailboxMessageFixture.MESSAGE_2;
+import static org.apache.james.mailbox.backup.MailboxMessageFixture.MESSAGE_CONTENT_1;
+import static org.apache.james.mailbox.backup.MailboxMessageFixture.MESSAGE_CONTENT_2;
+import static org.apache.james.mailbox.backup.MailboxMessageFixture.MESSAGE_ID_1;
+import static org.apache.james.mailbox.backup.MailboxMessageFixture.MESSAGE_ID_2;
+import static org.apache.james.mailbox.backup.ZipAssert.assertThatZip;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+
+import javax.mail.Flags;
+import javax.mail.util.SharedByteArrayInputStream;
+
+import org.apache.commons.compress.archivers.zip.ZipFile;
+import org.apache.james.junit.TemporaryFolderExtension;
+import org.apache.james.junit.TemporaryFolderExtension.TemporaryFolder;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.TestId;
+import org.apache.james.mailbox.model.TestMessageId;
+import org.apache.james.mailbox.store.mail.model.impl.PropertyBuilder;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailboxMessage;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import com.google.common.collect.ImmutableList;
+
+@ExtendWith(TemporaryFolderExtension.class)
+public class ZipperTest {
+    private Zipper testee;
+    private File destination;
+
+    @BeforeEach
+    void beforeEach(TemporaryFolder temporaryFolder) throws Exception {
+        testee = new Zipper();
+        destination = File.createTempFile("backup-test", ".zip", temporaryFolder.getTempDir());
+    }
+
+    @Test
+    void archiveShouldWriteEmptyValidArchiveWhenNoMessage() throws Exception {
+        testee.archive(ImmutableList.of(), new FileOutputStream(destination));
+
+        try (ZipFile zipFile = new ZipFile(destination)) {
+            assertThatZip(zipFile).hasNoEntry();
+        }
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
```

Comment:
also unneeded

Reply:
Need with IntelliJ :trollface:

Reply:
too bad, because it introduces me a "Unnecessary annotation" on Eclipse.
Anyway who looks at IntelliJ warnings? :)

Reply:
> Need with IntelliJ 

Maybe due to changes I did, but it's not needed in intellij now